Tests/BasicTests/SortedArrayTests.swift

```diff
@@ -30,9 +30,55 @@ class SortedArrayTests: XCTestCase {
 
         arr += [2, 3]
         XCTAssertEqual(arr.values, [-13, 0, 0, 1, 2, 2, 3, 3, 9, 13, 14, 15, 100, 198])
+        
+        arr.insert(-13)
+        XCTAssertEqual(arr.values, [-13, -13, 0, 0, 1, 2, 2, 3, 3, 9, 13, 14, 15, 100, 198])
+        
+        arr.insert(198)
+        XCTAssertEqual(arr.values, [-13, -13, 0, 0, 1, 2, 2, 3, 3, 9, 13, 14, 15, 100, 198, 198])
     }
+    
+    func testSortedArrayInDescendingOrder() throws {
+        var arr = SortedArray<Int>(areInIncreasingOrder: >)
+        arr.insert(15)
+        arr.insert(14)
+        arr.insert(100)
+        arr.insert(-13)
+        arr.insert(0)
+        arr.insert(198)
+        arr.insert(13)
+        XCTAssertEqual(arr.values, [198, 100, 15, 14, 13, 0, -13])
+        
+        arr.insert(contentsOf: [2, 1, 3, 0, 9])
+        XCTAssertEqual(arr.values, [198, 100, 15, 14, 13, 9, 3, 2, 1, 0, 0, -13])
+        
+        arr += [2, 3]
+        XCTAssertEqual(arr.values, [198, 100, 15, 14, 13, 9, 3, 3, 2, 2, 1, 0, 0, -13])
+        
+        arr.insert(-13)
+        XCTAssertEqual(arr.values, [198, 100, 15, 14, 13, 9, 3, 3, 2, 2, 1, 0, 0, -13, -13])
+        
+        arr.insert(198)
+        XCTAssertEqual(arr.values, [198, 198, 100, 15, 14, 13, 9, 3, 3, 2, 2, 1, 0, 0, -13, -13])
 
+
+    }
+
+    func testPerformanceOfSortedArrayInAscendingOrder() {
```

Comment:
This should be placed in a new file under `BasicPerformanceTests`, see `SHA256PerfTests.swift` for example.